time time
int32 tl
int32 a
